pkg/distributor/ha_tracker.go

```diff
@@ -402,8 +407,20 @@ func (h *haTracker) cleanupOldReplicas(ctx context.Context, deadline time.Time)
 	}
 }
 
+// ReplicaChecker expose checkReplica private method and add counter to track how many push request that is not coming
+// from elected replica.
+func (h *haTracker) ReplicaChecker() func(ctx context.Context, userID, cluster, replica string, now time.Time) error {
+	return func(ctx context.Context, userID, cluster, replica string, now time.Time) error {
+		err := h.checkReplica(ctx, userID, cluster, replica, now)
+		if err != nil && errors.Is(err, util.ReplicasNotMatchError{}) {
+			h.haPushRequestSkipParse.Inc()
+		}
+		return err
+	}
+}
```

Comment:
If this method is going to take the responsibility of increasing the `haPushRequestSkipParse` metric, then it should reflect that in the name. 
Following my other suggestion, I'd call this method as:

```go
func (*haTracker) shouldShortcutRequest(ctx context.Context, w http.ResponseWriter, req *http.Request) (finished bool) {
        // implementation from the other comment
```